pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -89,35 +89,58 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundHTTPRouteConfig(env *mo
 // buildSidecarOutboundHTTPRouteConfig builds an outbound HTTP Route for sidecar.
 // Based on port, will determine all virtual hosts that listen on the port.
 func (configgen *ConfigGeneratorImpl) buildSidecarOutboundHTTPRouteConfig(env *model.Environment, node *model.Proxy, push *model.PushContext,
-	proxyInstances []*model.ServiceInstance, services []*model.Service, routeName string) *xdsapi.RouteConfiguration {
+	proxyInstances []*model.ServiceInstance, routeName string) *xdsapi.RouteConfiguration {
 
 	listenerPort := 0
-	if routeName != RDSHttpProxy {
-		var err error
-		listenerPort, err = strconv.Atoi(routeName)
-		if err != nil {
+	var err error
+	listenerPort, err = strconv.Atoi(routeName)
+	if err != nil {
+		// we have a port whose name is http_proxy or unix:///foo/bar
+		// check for both.
+		if routeName != RDSHttpProxy || !strings.HasPrefix(routeName, model.UnixAddressPrefix) {
 			return nil
 		}
 	}
 
+	// Get the list of services that correspond to this egressListener from the sidecarScope	
+	sidecarScope := push.GetSidecarScope(node, proxyInstances)
```

Comment:
Same comments as before: this is changing the code paths for non-sidecar uses, which is extremely risky.

If Sidecar is not used - GetSidecarScope should return nil, and the old code should be used. 
If later on we want to refactor or improve the code non using Sidecar - we can do it, but it's impossible to understand all implications of this PR.

Reply:
See earlier comment on config == nil check